io-classes/src/Control/Monad/Class/MonadMVar.hs

```diff
@@ -344,9 +409,8 @@ isEmptyMVarDefault (MVar tv) =
     atomically $ do
       s <- readTVar tv
       case s of
-        MVarFull {} -> return False
-        MVarEmpty blockedq | null blockedq -> return True
-                           | otherwise -> error "isEmptyMVarDefault: invariant violation"
+        MVarFull  {} -> return False
+        MVarEmpty {} -> return True
```

Comment:
Related to my last comment, but I can think of scenarios where we encounter `MVarEmpty` with non-empty `readq`s and/or `takeq`s. Are they also as unlikely as mentioned in `takeMVar` and `putMVar`?

Reply:
I don't think that's unlikely at all. I think this code was just a mistake.